R/seq_linter.R

```diff
@@ -57,10 +67,18 @@ seq_linter <- function() {
       "seq_along",
       "seq_len"
     )
-    lint_message <- sprintf(
-      "%s:%s is likely to be wrong in the empty edge case. Use %s() instead.",
-      dot_expr1, dot_expr2, replacement
-    )
+
+    if (any(grepl("seq", dot_expr1, fixed = TRUE))) {
```

Comment:
should this be an `ifelse`? what happens for `c(seq(length(x)), 1:nrow(x))`?

Reply:
Good catch! Fixed it, and added a test as well:

``` r
library(lintr)

lint(
  text = "c(seq(length(x)), 1:nrow(y))",
  linters = seq_linter()
)
#> <text>:1:3: warning: [seq_linter] seq(length(...)) is likely to be wrong in the empty edge case. Use seq_along(...) instead.
#> c(seq(length(x)), 1:nrow(y))
#>   ^~~~~~~~~~~~~~
#> <text>:1:19: warning: [seq_linter] 1:nrow(...) is likely to be wrong in the empty edge case. Use seq_len() instead.
#> c(seq(length(x)), 1:nrow(y))
#>                   ^~~~~~~~~
```

<sup>Created on 2022-07-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>